app-modules/interaction/tests/Models/Scopes/InteractionConfidentialScopeTest.php

```diff
@@ -35,55 +35,92 @@
+    ]);
+
+    Interaction::factory()->count(10)->create([
+        'is_confidential' => true,
+    ]);
 
+    Interaction::factory()->count(10)->create([
+        'is_confidential' => false,
+    ]);
+
+    expect(Interaction::query()->count())->toBe(20);
+});
+
+test('Interactions model with fetch data count for team user', function () {
     $teamUser = User::factory()->licensed(LicenseType::cases())->create();
     $teamUser->givePermissionTo('interaction.view-any');
 
     $team = Team::factory()->hasAttached($teamUser, [], 'users')->create();
 
-    $confidentialInteraction = Interaction::factory()->hasAttached($user, [], 'confidentialAccessUsers')->hasAttached($team, [], 'confidentialAccessTeams')->count(10)->create([
+    actingAs($teamUser);
+    Interaction::factory()->hasAttached($team, [], 'confidentialAccessTeams')->count(10)->create([
         'is_confidential' => true,
     ]);
 
-    $nonConfidentialInteraction = Interaction::factory()->count(10)->create([
+    Interaction::factory()->count(10)->create([
+        'is_confidential' => true,
+    ]);
+
+    Interaction::factory()->count(10)->create([
         'is_confidential' => false,
     ]);
 
-    $allInteractions = $confidentialInteraction->merge($nonConfidentialInteraction);
+    expect(Interaction::query()->count())->toBe(20);
+});
+
+test('Interactions model with fetch data count for assigned user', function () {
+    $user = User::factory()->licensed(LicenseType::cases())->create();
+    $user->givePermissionTo('interaction.view-any');
 
     actingAs($user);
-    livewire(ListInteractions::class)
-        ->set('tableRecordsPerPage', 20)
-        ->assertCanSeeTableRecords($allInteractions);
+    Interaction::factory()->hasAttached($user, [], 'confidentialAccessUsers')->count(10)->create([
+        'is_confidential' => true,
+    ]);
 
-    actingAs($userWithoutAttach);
-    livewire(ListInteractions::class)
-        ->set('tableRecordsPerPage', 10)
-        ->assertCanSeeTableRecords($nonConfidentialInteraction)
-        ->assertCanNotSeeTableRecords($confidentialInteraction);
+    Interaction::factory()->count(10)->create([
+        'is_confidential' => true,
+    ]);
 
-    actingAs($teamUser);
-    livewire(ListInteractions::class)
-        ->set('tableRecordsPerPage', 20)
-        ->assertCanSeeTableRecords($allInteractions);
+    Interaction::factory()->count(10)->create([
+        'is_confidential' => false,
+    ]);
+
+    expect(Interaction::query()->count())->toBe(20);
+});
 
+test('Interactions model with fetch data count for superadmin user', function () {
     asSuperAdmin();
-    livewire(ListInteractions::class)
-        ->set('tableRecordsPerPage', 20)
-        ->assertCanSeeTableRecords($allInteractions);
+    Interaction::factory()->count(10)->create([
+        'is_confidential' => true,
+    ]);
+
+    Interaction::factory()->count(10)->create([
+        'is_confidential' => false,
+    ]);
+
+    expect(Interaction::query()->count())->toBe(20);
```

Comment:
Would shouldn't just check count. We should also be checking the actual records that are returned to see if they are the records we expect.